### ISSUE TITLE
Start of line not matching correctly?

### ISSUE THREAD

**Issue by mbucc** [martanne/vis#264]
Using sam from 9fans/plan9port, I get this:

```
$ sam -d t
 -. t
,p
line1
line2
, x/^/c/#/
,p
#line1
#line2
```

When I try the command `, x/^/c/#/` in vis (built from HEAD today), I get this:

```
#li#ne#1
#li#ne#2
```

I expected the vis output to match what sam produced.

**Comment by martanne** [comment]
Thanks for the report! I'm especially interested in bug reports from sam/acme users regarding the structural regular expression support.

Part of the problem was that apparently [Plan 9's regexp library](https://swtch.com/plan9port/man/man7/regexp.html) and [POSIX regexec](http://pubs.opengroup.org/onlinepubs/009695399/functions/regcomp.html) when used with `REG_NEWLINE` have different semantics for `^`.

I'm not too happy with the fix, because it might cause other issues. Maybe I should explicitly test for an `\n` character before the last empty match.

Anyway the following seems to work:

```
x/^/c/#
x i/#
y/\n/i/#
```

**Comment by mbucc** [comment]
I confirmed this is working for me.  Thanks for the quick fix!

**Comment by martanne** [comment]
The last example I gave `y/\n/i/#` should actually behave differently. The empty match at the end of the file (after the terminating newline) should also be looped over, thus adding one more `#` than the other commands, resulting in:
```
#line1
#line2
#
```
This was fixed in fb414a7a85c6b8843d18030f07db0b8c80cc0251.
